transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionSubsystemAdd.java

```diff
@@ -473,6 +474,14 @@ private void performRecoveryEnvBoottime(OperationContext context, ModelNode mode
                 .addService(TxnServices.JBOSS_TXN_EXTENDED_JBOSS_XA_TERMINATOR, extendedJBossXATerminatorService)
                 .setInitialMode(Mode.ACTIVE).install();
 
+        final JBossContextXATerminatorService contextXATerminatorService = new JBossContextXATerminatorService();
+        context.getServiceTarget()
+                .addService(TxnServices.JBOSS_TXN_CONTEXT_XA_TERMINATOR, contextXATerminatorService)
```

Comment:
I am wondering if transactionManagerServiceServiceBuilder should have a dependency on JBOSS_TXN_CONTEXT_XA_TERMINATOR instead of JBOSS_TXN_XA_TERMINATOR - it's not my area of specialism but I am just trying to work through it logically as to what you introduced and what it "replaces"/"augments" and how